Bump devel tag.py for stable-2.17

oraNod opened the pull request:
Resolves #1246 by updating the devel version to 2.17 / Ansible 10.

oraNod committed:
update tagger for 2.17

oraNod commented on docs/docsite/rst/conf.py:
```diff
@@ -238,15 +238,15 @@
     # list specifically out of order to make latest work
     'available_versions': (
         ('devel',) if tags.has('all') else
         ('2.15_ja', '2.14_ja', '2.13_ja',) if tags.has('core_lang') else
```
@samccann I guess we need to leave this line unchanged as 2.15 is the last version that includes a Japanese translation. Does that make sense?

samccann replied:
So two things going on here:
1 - I forgot to EOL the japanese 2.13, so this could go in a separate PR and be merged, then backported all the way down to 2.13... then we republish the japanese docs 2-13 through 2.15.

2. We then repeat this effort after 2.17 releases, to remove 2.14 similarly from the japanese docs.

My nickel would be to just bunch them both together in a separate PR and don't merge it til release day (and then backport all the way down to 2.13 and republish.

We'd also have to EOL 2.13 and 2.14 japanese docs... That's a separate PR that only goes into those individual stable branches.

oraNod replied:
I've created a separate issue for this so we don't forget it: https://github.com/ansible/ansible-documentation/issues/1255

oraNod commented on docs/docsite/rst/conf.py:
```diff
@@ -238,15 +238,15 @@
-        '2.16' if tags.has('core_lang') or tags.has('core') else
-        '9' if tags.has('ansible')
+        '2.17' if tags.has('core_lang') or tags.has('core') else
+        '10' if tags.has('ansible')
```
@samccann I didn't do it when I created the `stable-2.17` branch but we'll need to make the same changes to the conf.py there as well, isn't that so?

samccann replied:
We can't merge the 2.17 change until core-2.17 releases (see checklist - https://github.com/ansible/ansible-documentation/issues/1249#issuecomment-2037569231)

It also has to be made in two separate PRs. Core becomes 2.17 a month or so before Ansible becomes Ansible 10.  (And similarly, the Ansible 10 change can't be merged until Ansible 10 release day).

Changing earlier than the release day for each of these will mess up either the version switcher or the label of the version above it (I forget which off the top of my head).

gotmax23 commented on hacking/tagger/tag.py:
```diff
@@ -41,7 +41,7 @@
 DEFAULT_ANSIBLE_CORE_CHECKOUT = ROOT.parent.joinpath("ansible")
 DEFAULT_REMOTE = "origin"
-DEFAULT_ACTIVE_BRANCHES: tuple[str, ...] = ("stable-2.14", "stable-2.15", "stable-2.16")
+DEFAULT_ACTIVE_BRANCHES: tuple[str, ...] = ("stable-2.15", "stable-2.16", "stable-2.17")
```
```suggestion
DEFAULT_ACTIVE_BRANCHES: tuple[str, ...] = ("stable-2.14", "stable-2.15", "stable-2.16", "stable-2.17")
```

oraNod replied:
Thanks @gotmax23 you're right that 2.14 is still active until Ansible 10 goes out.

gotmax23 replied:
We want to keep this here until ansible-core 2.14.z goes EOL. The tags should be created whether or not the corresponding community package version is still supported.

oraNod committed:
Update hacking/tagger/tag.py

Co-authored-by: Maxwell G <maxwell@gtmx.me>

oraNod committed:
format tag.py

samccann commented on docs/docsite/rst/conf.py:
```diff
@@ -238,15 +238,15 @@
         ('devel',) if tags.has('all') else
         ('2.15_ja', '2.14_ja', '2.13_ja',) if tags.has('core_lang') else
-        ('2.16', '2.15', '2.14', 'devel',) if tags.has('core') else
+        ('2.17', '2.16', '2.15', 'devel',) if tags.has('core') else
```
Can't make this change until release day as well, but it could go in the same PR that changes line 241 (for core 2.17 release day).

samccann commented on hacking/tagger/tag.py:
```diff
@@ -41,7 +41,12 @@
+    "stable-2.15",
+    "stable-2.16",
+    "stable-2.17",
+)
```
This one is a bit beyond me, but based on the title (DEFAULT ACTIVE BRANCHES) - it seems safe to merge immediately. But I'd get @gotmax23 review here to be sure.

Then after 2.14 EOLs I'd guess we remove that one?

oraNod replied:
See the comment here: https://github.com/ansible/ansible-documentation/pull/1251#discussion_r1552276502

samccann reviewed (commented):
tbh - I follow the release checklists to the letter because I get confused on all the different versions and when to fire off each change.  There may be an easier way to do it, but this particular PR has some problems in terms of timing of the changes and when each can be merged. I added notes, but in general, we have to merge some only on core-2.17 release day, and merge others only on Ansible 10 release day.

oraNod commented:
> tbh - I follow the release checklists to the letter because I get confused on all the different versions and when to fire off each change. There may be an easier way to do it, but this particular PR has some problems in terms of timing of the changes and when each can be merged. I added notes, but in general, we have to merge some only on core-2.17 release day, and merge others only on Ansible 10 release day.

@samccann Thanks, I guess I was getting ahead of myself with some of the changes. I've dropped the commit to modify version in the config. We'll probably want to keep the updates to the tag.py script though.
